core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -105,6 +105,18 @@ package object config {
     .bytesConf(ByteUnit.MiB)
     .createOptional
 
+  private[spark] val DRIVER_MEMORY_OVERHEAD_FACTOR =
+    ConfigBuilder("spark.driver.memoryOverheadFactor")
+      .doc("Fraction of driver memory to be allocated as additional non-heap memory per driver " +
+        "process in cluster mode. This is memory that accounts for things like VM overheads, " +
+        "interned strings, other native overheads, etc. This tends to grow with the container " +
+        "size. This value is ignored if spark.driver.memoryOverhead is set directly.")
```

Comment:
Right so what I meant here is that since we deprecated  spark.kubernetes.memoryOverheadFactor we should remove it from the running-on-kubernetes documentation and then the description of 0.40 for non-JVM jobs needs to move here.  ie On Kubernetes the default is 0.10 for JVM-based jobs and 0.40 for non-JVM jobs.

Reply:
Updated the doc to describe the kubernetes non-jvm default. Also removed the kubernetes config from the docs